services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -20,7 +18,8 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
   ): AsyncResult[List[String]] = ???
 
   override def createRepository(project: Project)(implicit ec: ExecutionContext): AsyncResult[Project] = {
-    def projectWithRepository(repositoryFullPath: String): Project = project.copy(repository = repositoryFullPath)
+    def projectWithRepository(repositoryFullPath: String): Project =
+      project.copy(repository = Some(Repository(repositoryFullPath)))
```

Comment:
It should be part of `Project` entity in my opinion. 
So it should be defined inside `Project` case class with the name `withRepository`

Reply:
Done. Please check, did I understand you right.

Reply:
@AndreiMaksimenko It should take Option[String]. 
In your case if we pass there, for example, `null` it won't be processed nicely. (Because of the `Some` )

Reply:
@ilyaonishenko are you sure that it makes sense here? If `repositoryFullPath` is `null`, then we will have the following
`Option(Repository(null))`
not
`Option(null)`
Therefore, `Option` will not save us anyway.

Reply:
Hmmm... it is strange I think. In that case of null who guarantees that there will not be Some[Null] in that Option - and we will have the same situation.

Reply:
I can add one more method withNoRepositry() - to create empty repo Project.

Reply:
Fixed